File: app/src/main/java/com/mv2studio/myride/connection/CommandState.java
package com.mv2studio.myride.connection;

import android.support.annotation.IntDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

/**
 * Created by matej on 30.10.16.
 */

@Retention(RetentionPolicy.SOURCE)
@IntDef({CommandState.NEW, CommandState.RUNNING, CommandState.FINISHED,
        CommandState.EXECUTION_ERROR, CommandState.BROKEN_PIPE,
        CommandState.QUEUE_ERROR, CommandState.NOT_SUPPORTED})
public @interface CommandState {

    int NEW = 0;
    int RUNNING = 1;
    int FINISHED = 2;
    int EXECUTION_ERROR = 3;
    int BROKEN_PIPE = 4;
    int QUEUE_ERROR = 5;
    int NOT_SUPPORTED = 6;

}

File: app/src/main/kotlin/com/mv2studio/myride/connection/CommandState.kt
package com.mv2studio.myride.connection

import android.support.annotation.IntDef
import kotlin.annotation.Retention

/**
 * Created by matej on 15/11/2016.
 */
@Retention(AnnotationRetention.RUNTIME)
@IntDef(State.NEW, State.RUNNING, State.FINISHED,
    State.EXECUTION_ERROR, State.BROKEN_PIPE,
    State.QUEUE_ERROR, State.NOT_SUPPORTED)
annotation class CommandState